### PR TITLE
Assorted updates

### PR DESCRIPTION
**Pull request by jbq**
- move to Java 8
- upgrade protobuf dependency to version 3.3.1
- bump version number

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
   <groupId>com.hubspot</groupId>
   <artifactId>basepom</artifactId>
-  <version>15.14-SNAPSHOT</version>
+  <version>15.15-SNAPSHOT</version>
```
Comment (stevegutz) [inline pom.xml]: We have our own internal release system that increments the versions for us.

File: pom.xml [hunk 2]
```diff
@@ -109,6 +109,8 @@
     <dep.swagger.version>1.3.12</dep.swagger.version>
     <dep.swagger-plugin.version>2.3.2</dep.swagger-plugin.version>
 
+    <dep.protobuf.version>3.3.1</dep.protobuf.version>
```
Comment (stevegutz) [inline pom.xml]: We're currently on an older versions of protobuf and the upgrade process is nontrivial for large, distributed codebases.

File: pom.xml [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
   <properties>
     <project.jdk7.home>${env.JAVA_HOME}</project.jdk7.home>
-    <project.build.targetJdk>1.7</project.build.targetJdk>
+    <project.build.targetJdk>1.8</project.build.targetJdk>
```
Comment (stevegutz) [inline pom.xml]: Some of our more widely-used opensource projects are also still on Java 7 and I'm not sure that we want to upgrade all of them beyond what downstream consumers may be using.

**Comment by jbq** [comment]
Then you could bump the major version number perhaps?
-- 
Jean-Baptiste Quenot
Tel maison: +33(0)5 62 57 17 28
Tel portable: +33(0)6 50 23 76 87

**Comment by stevegutz** [comment]
The major version corresponds to the version of `org.basepom:basepom-minimal-oss` we're using as a parent.

**Comment by jbq** [comment]
Then if basepom is not supposed to change and stay on old versions, I will try to do those changes in `jackson-datatype-protobuf` directly.

**Pull request closed without merging**